[python] reintroduce pympler

truthbk opened the pull request:
### What does this PR do?

This PR reintroduces pympler to provide python memory space snapshots and help us identify potential object leaks, etc.

### Motivation

Need for insights on six and python memory spaces. We need multiple tools to identify various potential sources of memory mismanagement.

### Additional Notes

Extends the `status` command to allow querying individual components available via the API, in our particular case of interest: `python`.

truthbk committed:
[py][six] adding pympler interpreter memory usage stats

[six][two] fixing typo

truthbk committed:
[dep] updating gopkg.lock

truthbk committed:
[agent][api] fix python memory status call

[agent][api] add python build flag

[agent][api] add getPythonStatus stub when python build is not enabled

[python] fix pympler reporting - use structs for JSON structure

[agent][cmd] make it easier to query component status - python stats

remeh commented on cmd/agent/app/status.go:
```diff
@@ -49,32 +52,37 @@ var statusCmd = &cobra.Command{
+			color.NoColor = true
+		}
+
+		if len(args) != 1 {
```
Detail but seeing the usage:
```
	Use:   "status [component [name]]",
```
there is two parameters possible, so it should be `< 2`.

remeh replied:
Whoops, got it, forget that.

remeh commented on six/three/three.cpp:
```diff
@@ -762,3 +762,51 @@ char *Three::getIntegrationList()
+        goto done;
+    }
+
+    memUsage = as_yaml(summary);
```
You didn't check `if memUsage == null` in three (compared to two).

remeh reviewed (dismissed):
Left small comments.

truthbk committed:
[release] build off Maxime guppy branch

albertvaka commented on cmd/agent/api/agent/python_status.go:
```diff
@@ -7,15 +7,26 @@
 // This group of endpoints is meant to provide high-level functionalities
 // at the agent level.
 
+// +build python
```
Should we disable it on Windows since it was crashing there?

```
// +build python !windows
```

truthbk replied:
I intended to test on windows :) Let me double-check that before we remove windows builds :)

albertvaka commented on cmd/agent/api/agent/python_status_nopython.go:
```diff
@@ -0,0 +1,20 @@
+// This group of endpoints is meant to provide high-level functionalities
+// at the agent level.
+
+// +build !python
```
```
build !python, windows
```

truthbk committed:
[pympler] disable on windows for now

truthbk committed:
[rtloader] fix six naming vestiges

[rtloader][two] old six type renamed to rtloader

[rtloader][three] old six type renamed to rtloader

[rtloader][cgo] fix pympler helper from depcreated six naming

truthbk committed:
Gopkg.lock was out of sync

truthbk committed:
fix nightly build - integrations from master
